### ISSUE TITLE
UseEqualsToCompareStrings should consider return type

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3843]
**Affects PMD Version:6.42.0** 

**Rule:UseEqualsToCompareStrings**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_errorprone.html#useequalstocomparestrings>

**Description:**
Hi, I found a false negative about the rule **UseEqualsToCompareStrings**. Please review the following code example. PMD should have reported a warning at line 5 because the string x is compared with another string, but it doesn't. Hence, I think it is a false negative.

**Code Sample demonstrating the issue:**

```java
String getLiteral() {
  return "hello";
}
void UseEqualsToCompareStrings(String x) {
  if (x == getLiteral()) {} // a false negative should report
}
```

**Expected outcome:**

PMD should report a violation at line 5, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by jsotuyod** [comment]
This is fixed in PMD 7.0.0-rc1. It won't be backported to PMD 6.
